internal/controller/storage/storage.go

```diff
@@ -0,0 +1,95 @@
+package storage
+
+import (
+	"context"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/util/retry"
+
+	danaiov1alpha1 "github.com/dana-team/nfspvc-operator/api/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	nfsPvcDanaLabel = "nfspvc.dana.io/nfspvc-owner"
+)
+
+func PreparePVC(nfspvc danaiov1alpha1.NfsPvc, StorageClass string) corev1.PersistentVolumeClaim {
+	storageClass := StorageClass
+	return corev1.PersistentVolumeClaim{
+		TypeMeta: metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      nfspvc.Name,
+			Namespace: nfspvc.Namespace,
+			Labels: map[string]string{
+				nfsPvcDanaLabel: nfspvc.Name,
+			},
+		},
+		Spec: corev1.PersistentVolumeClaimSpec{
+			StorageClassName: &storageClass,
+			VolumeName:       nfspvc.Name + "-" + nfspvc.Namespace + "-pv",
+			AccessModes:      nfspvc.Spec.AccessModes,
+			Resources: corev1.VolumeResourceRequirements{
+				Requests: nfspvc.Spec.Capacity,
+			},
+		},
+	}
+}
+
+func PreparePV(nfspvc danaiov1alpha1.NfsPvc, StorageClass string, ReclaimPolicy string) corev1.PersistentVolume {
+	return corev1.PersistentVolume{
+		TypeMeta: metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: nfspvc.Name + "-" + nfspvc.Namespace + "-pv",
+			Labels: map[string]string{
+				nfsPvcDanaLabel: nfspvc.Name,
+			},
+		},
+		Spec: corev1.PersistentVolumeSpec{
+			StorageClassName:              StorageClass,
+			Capacity:                      nfspvc.Spec.Capacity,
+			AccessModes:                   nfspvc.Spec.AccessModes,
+			PersistentVolumeReclaimPolicy: corev1.PersistentVolumeReclaimPolicy(ReclaimPolicy),
+			ClaimRef: &corev1.ObjectReference{
+				Name:      nfspvc.Name,
+				Namespace: nfspvc.Namespace,
+				Kind:      corev1.ResourcePersistentVolumeClaims.String(),
+			},
+			PersistentVolumeSource: corev1.PersistentVolumeSource{
+				NFS: &corev1.NFSVolumeSource{
+					Server: nfspvc.Spec.Server,
+					Path:   nfspvc.Spec.Path,
+				},
+			},
+		},
+	}
+}
+
+// UpdatePV updates a PV if the nfspvc is updated.
+func UpdatePV(ctx context.Context, nfspvc danaiov1alpha1.NfsPvc, k8sClient client.Client, pv corev1.PersistentVolume) error {
```

Comment:
The function makes no conditional on whether the `nfspvc` is updated or not. This logic is on the caller side.